backend/drm/atomic.c

```diff
@@ -140,16 +140,19 @@ static bool atomic_conn_enable(struct wlr_drm_backend *drm,
 		return !enable;
 	}
 
+	/* crtc will be enabled on next page flip */
+	if (enable) {
+		crtc->needs_modeset = true;
```

Comment:
This seems to set `DRM_MODE_ATOMIC_ALLOW_MODESET`. However we already set it in the `atomic_commit` call, so I don't think it's necessary.

Reply:
it's for the case where we `enable` the crtc but `atomic_crtc_pageflip()` does not get passed a mode.